### PR TITLE
Add Metrics

### PR DESCRIPTION
**Pull request by jrconlin**
- Fork server/bleed_serve for Moz specific changes
  *\* Add metric tracking. This is statsd supporting as well as stores a local instance snapshot.
  *\* Some work done to shore up arguments (default values, et. al.)

Notes:
I've tried to stick to the same model as the original code where possible. It might be useful to diff server/bleed_serve.go to see actual changes. 
The metric handler is an anonymous function in order to keep the code footprint smaller.
Kind of not a fan of docopt, mostly because of the way go doesn't handle interface dereferencing.

### CODE REVIEW

**Comment by oremj** [comment]
It looks like we aren't really modifying the original codebase, just adding our own main. What do you think about starting a new project, with our own main module, which imports his repo? Or, do you think it is worth it to maintain a complete fork?

**Comment by jrconlin** [comment]
Part of it is the way that the original code is structured and how go wants to package things. Since all of the handlers are in the "main" package, and since "main" is the root package for the code, it's hard to import the handlers directly. Creating a wholly different fork also would make contributing things back to the main branch difficult.

By creating a forked server file, we can still pull from the original repo and quickly modify the code to merge the differences. (This patch effectively touches two of the original files.)

Once I get a second set of eyes on this, I'm going to send two PRs to the original fork, one that includes the argument fixes for usable defaults, and another for the optional metrics info.

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/cactus/go-statsd-client/statsd"
+)
+
+type Metrics struct {
+	sync.Mutex
+	dict   map[string]int64   // counters
```
Comment (crankycoder) [inline server/metrics/metrics.go:19]: Minor nit.  The metrics code for the dict and timer maps assumes that the caller is well behaved (which in this case it is) and isn't going to be polluting the map and just eat all the memory.

Do we really need to be dumping metrics out?  Is statsd really insufficient?

**Comment by crankycoder** [comment]
@jrconlin  r+ 
Everything else looks pretty much the same from the original server/bleed_serve.go code - so there's not much to nitpick on.

**Comment by oremj** [comment]
This PR is included in #4

**Pull request merged**